cmd/cli/commands/integration_test.go

```diff
@@ -0,0 +1,452 @@
+	ref  string
+}
+
+// generateReferenceTestCases generates a comprehensive list of reference formats for testing
+func generateReferenceTestCases(info modelInfo) []referenceTestCase {
+	cases := []referenceTestCase{
+		{
+			name: "short form (no registry, no org, no tag)",
+			ref:  info.name,
+		},
+		{
+			name: "with tag",
+			ref:  fmt.Sprintf("%s:%s", info.name, info.tag),
+		},
+		{
+			name: "with org",
+			ref:  fmt.Sprintf("%s/%s", info.org, info.name),
+		},
+		{
+			name: "with org and tag",
+			ref:  fmt.Sprintf("%s/%s:%s", info.org, info.name, info.tag),
+		},
+		{
+			name: "fqdn",
+			ref:  fmt.Sprintf("%s/%s/%s:%s", info.registry, info.org, info.name, info.tag),
+		},
+		{
+			name: "with registry and org",
+			ref:  fmt.Sprintf("%s/%s/%s", info.registry, info.org, info.name),
+		},
+		{
+			name: "by digest",
+			ref:  fmt.Sprintf("%s@%s", info.name, info.digest),
+		},
+		{
+			name: "by digest with org",
+			ref:  fmt.Sprintf("%s/%s@%s", info.org, info.name, info.digest),
+		},
+		{
+			name: "by digest with registry and org",
+			ref:  fmt.Sprintf("%s/%s/%s@%s", info.registry, info.org, info.name, info.digest),
+		},
+		{
+			name: "full model ID",
+			ref:  info.modelID,
+		},
+		{
+			name: "truncated model ID (12 chars)",
+			ref:  info.modelID[7:19],
+		},
+		{
+			name: "model ID without sha256 prefix",
+			ref:  strings.TrimPrefix(info.modelID, "sha256:"),
+		},
+	}
+
+	return cases
+}
+
+// setupTestEnv creates the complete test environment with registry and DMR
+func setupTestEnv(t *testing.T) *testEnv {
+	ctx := context.Background()
+
+	// Create a custom network for container communication
+	net, err := network.New(ctx)
+	require.NoError(t, err)
+	testcontainers.CleanupNetwork(t, net)
+
+	registryURL := ociRegistry(t, ctx, net)
+	dmrURL := dockerModelRunner(t, ctx, net)
+
+	modelRunnerCtx, err := desktop.NewContextForTest(dmrURL, nil)
+	require.NoError(t, err, "Failed to create model runner context")
+
+	client := desktop.New(modelRunnerCtx)
+	if !client.Status().Running {
+		t.Fatal("DMR is not running")
+	}
+
+	return &testEnv{
+		ctx:         ctx,
+		registryURL: registryURL,
+		client:      client,
+	}
+}
+
+func ociRegistry(t *testing.T, ctx context.Context, net *testcontainers.DockerNetwork) string {
+	t.Log("Starting OCI registry container...")
+	ctr, err := testcontainers.Run(
+		ctx, "registry:3",
+		testcontainers.WithExposedPorts("5000/tcp"),
+		testcontainers.WithWaitStrategy(wait.ForHTTP("/v2/").WithPort("5000/tcp").WithStartupTimeout(30*time.Second)),
+		network.WithNetwork([]string{"registry.local"}, net),
+	)
+	require.NoError(t, err)
+	testcontainers.CleanupContainer(t, ctr)
+
+	registryEndpoint, err := ctr.Endpoint(ctx, "")
+	require.NoError(t, err)
+	registryURL := fmt.Sprintf("http://%s", registryEndpoint)
+	t.Logf("Registry available at: %s", registryURL)
+	return registryURL
+}
+
+func dockerModelRunner(t *testing.T, ctx context.Context, net *testcontainers.DockerNetwork) string {
```

Comment:
suggestion: use the DMR module

Reply:
Oh I see, you need the in-code DMR, not the one installed in DD. Thanks for the clarification!